### PR TITLE
Integrate with a geolocation API

### PR DESCRIPTION
**Pull request by BrunoTiba**

### CODE REVIEW

File: data/domain/src/main/java/com/brunotiba/domain/repository/WeatherRepository.kt [hunk 1]
```diff
@@ -2,7 +2,16 @@ package com.brunotiba.domain.repository
 
 import com.brunotiba.domain.model.Forecast
 
+/**
+ * Repository if location information.
```
Comment (igorescodro) [inline data/domain/src/main/java/com/brunotiba/domain/repository/WeatherRepository.kt]: I think this sentence has a typo.

File: data/remote/src/main/java/com/brunotiba/remote/api/LocationApi.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ */
+internal interface LocationApi {
+
+    /**
+     * Retrieves the location coordinates corresponding to the given name.
+     *
+     * @param name the location name
+     * @return a List containing the possible coordinates
+     */
+    @GET("search.php")
```
Comment (igorescodro) [inline data/remote/src/main/java/com/brunotiba/remote/api/LocationApi.kt:18]: PHP? Really?

File: data/remote/src/main/java/com/brunotiba/remote/api/LocationService.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    override val apiUrl: String
+        get() = BuildConfig.GEOLOCATION_API_URL
+
+    override fun getMandatoryParams(): List<QueryParameter> =
+        listOf(
+            QueryParameter(API_KEY_QUERY, BuildConfig.GEOLOCATION_API_KEY),
+            QueryParameter(API_FORMAT_QUERY, API_FORMAT_VALUE)
+        )
+
+    private val locationApi: LocationApi = getApi()
```
Comment (igorescodro) [inline data/remote/src/main/java/com/brunotiba/remote/api/LocationService.kt:23]: Suggestion: Wouldn't be better to have the result from getApi() stored in a lazy parameter inside the parent class? This way you can avoid having this private val in every implementation of ApiService and ensures that this snippet will be executed only once.

File: data/remote/src/main/java/com/brunotiba/remote/provider/RetrofitProvider.kt [hunk 4]
```diff
@@ -17,8 +18,8 @@ internal class RetrofitProvider(private val clientProvider: ClientProvider) {
      *
      * @return the Retrofit implementation
      */
-    fun getRetrofit(apiUrl: String): Retrofit {
-        val client = clientProvider.getClient()
+    fun getRetrofit(apiUrl: String, parameters: List<QueryParameter> = listOf()): Retrofit {
```
Comment (igorescodro) [inline data/remote/src/main/java/com/brunotiba/remote/provider/RetrofitProvider.kt:22]: Please, add new param in KDoc.

**Review by igorescodro — APPROVED** [review]
Just minor suggestions. Feel free to merge it.

File: data/remote/src/main/java/com/brunotiba/remote/api/LocationApi.kt (hunk 2 above)
Comment (BrunoTiba) [reply data/remote/src/main/java/com/brunotiba/remote/api/LocationApi.kt:18]: IKR

File: data/remote/src/main/java/com/brunotiba/remote/api/LocationService.kt (hunk 3 above)
Comment (BrunoTiba) [reply data/remote/src/main/java/com/brunotiba/remote/api/LocationService.kt:23]: This change would probably require a large effort for the obtained benefit.

**Pull request merged**